src/code/containers/challenge-container-selector.js

```diff
@@ -52,8 +56,10 @@ class ChallengeContainerSelector extends Component {
     if (routeParams.challengeId) {
       routeParams = AuthoringUtils.challengeIdToRouteParams(authoring, routeParams.challengeId);
     }
-    if (hasChangedRouteParams(this.props.currentRouteSpec, routeParams)) {
-      navigateToCurrentRoute({level: routeParams.level-1, mission: routeParams.mission-1, challenge: routeParams.challenge-1});
+    if (isValidRouteParams) {
```

Comment:
The defaulting path is inaccessible as written: `isValidRouteParams` is a function that must be called -- `isValidRouteParams()`. Beyond the defaulting issue, when I enter the navigate URL I end up with `routeParams` that has a `challengeId` but not a `level`, `mission`, or `challenge`, which then fails in a different way. I think the code needs some more work before it's worth reviewing further. Given that I'm going to be gone for the next couple of days, that means Sam will have to do further reviews.